packages/styles/scss/components/horizontalrule/_horizontalrule.scss

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright IBM Corp. 2016, 2018
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+@import '~@carbon/themes/scss/themes';
+@import '~carbon-components/scss/globals/scss/vars';
+@import '~carbon-components/scss/globals/scss/helper-mixins';
+@import '~carbon-components/scss/globals/scss/layout';
+@import '~carbon-components/scss/globals/scss/vendor/@carbon/elements/scss/import-once/import-once';
+@import '~carbon-components/scss/globals/scss/css--reset';
+
+$low-contrast: $ui-03;
+$medium-contrast: $ui-04;
+$high-contrast: $ui-05;
+
+@mixin horizontalrule {
+    .#{$prefix}--hr {
+        margin: $spacing-05 0;
+        border: none;
+        height: carbon--rem(1px);
+        background: $medium-contrast;
+    
+        &--small {
+            width: carbon--mini-units(4);
+        }
+    
+        &--medium {
+            width: carbon--mini-units(6);
+        }
+    
+        &--large {
+            width: carbon--mini-units(8);
+        }
+
+        &--low-contrast {
+            background: $low-contrast;
+        }
+    
+        &--high-contrast {
+            background: $high-contrast;
+        }
+    }
+
+    .#{$prefix}--hr--dashed.#{$prefix}--hr--fluid{
+        background: none;
+        background-size: $spacing-05 1px;
+        background-repeat: repeat-x;
+        background-image: linear-gradient(to right, $medium-contrast 50%, transparent 50%);
+
+        &.#{$prefix}--hr--low-contrast {
+            background-image: linear-gradient(to right, $low-contrast 50%, transparent 50%);
+        }
+
+        &.#{$prefix}--hr--high-contrast {
+            background-image: linear-gradient(to right, $high-contrast 50%, transparent 50%);
```

Comment:
I got a wild idea but hear me out on this... what if we set the `background-image` always whenever we set the `background-color`? It should just blend into the background on solid, but for dashed we could just remove the background color.

```scss
.#{$prefix}--hr {
  background-color: $medium-contrast;
  background-image: linear-gradient(to right, $medium-contrast 50%, transparent 50%);

  &--low-contrast {
    background-color: $low-contrast;
    background-image: linear-gradient(to right, $low-contrast 50%, transparent 50%);
  }

  &--high-contrast {
    background-color: $high-contrast;
    background-image: linear-gradient(to right, $high-contrast 50%, transparent 50%);
  }

  &--dashed {
    background-color: transparent;
    background-repeat: repeat-x;
    background-size: $spacing-05 $thin-border-weight;
    height: $thin-border-weight;
  }

  &--small,
  &--medium,
  &--large {
    background-image: none;
  }
}
```

The only drawback I'm seeing to this, and I haven't had enough time to see if there is a solution, but the `.bx--hr.bx--hr--small.bx--hr--dashed` provides no line. But you could make an argument that that's the desired outcome here, right? 😂

Just something to consider. This comment by no means should block this from being merged.